### ISSUE TITLE
[Analyzer] NullDereference false positive

### ISSUE THREAD

**Issue by jstengleingithub** [llvm/llvm-project#206798]
For this file, we see an incorrect Null warning:

https://godbolt.org/z/1Yjf4aGzj

Incorrect warning:
<source>:10:18: warning: Access to field 'n' results in a dereference of a null pointer (loaded from variable 'p') [clang-analyzer-core.NullDereference]

At line 10, p cannot be NULL due to the return at line 9

**Comment by chrchr-github** [comment]
Slightly reduced and as C++:
```c++
struct S { int n; };
struct T { S s; };

int g(S *p) {
    if (!p) {}
    if (nullptr == p)
        return 0;
    return p->n;
}

void f(T* r) {    
    g(&r->s);
}
```
https://godbolt.org/z/d1qEYYvvW
It looks like the inverted condition ` if (nullptr == p)` is not properly recognized.
